python/Ganga/GPIDev/Base/Objects.py

```diff
@@ -339,6 +377,19 @@ def getNodeIndexCache(self):
 ##########################################################################
 
 
+def synchronised_descriptor(get_or_set):
+    """
+    This decorator should only be used on ``__get__`` or ``__set__`` methods of a ``Descriptor``.
+    """
+    @functools.wraps(get_or_set)
+    def decorated(self, obj, type_or_value):
+        if obj is None:
```

Comment:
This check is needed since if a descriptor is [called on an class](https://docs.python.org/2/howto/descriptor.html#invoking-descriptors) (rather than an object) e.g. `f = LocalFile.subfiles`, the first argument passed to `__get__` is `None` and the second argument is the type. If we're requesting a value from a class then we don't want to lock the object (since we have none to lock).

Unless I misunderstand the question?